resources/js/tableofcontent.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * Construct recurseData.toc as a JSON-like that contains each header's text
+ * anchor, and their list of child headers.
+ * 
+ * References:
+ * https://stackoverflow.com/questions/187619/is-there-a-javascript-solution-to-generating-a-table-of-contents-for-a-page
+ * @param elem DOM element
+ * @param recurseData Object with fields: { toc : Object, count : int, parent : Object }
+ */
+function recurseChild(elem, recurseData)
```

Comment:
This (and any other global name introduced by this script) may conflict with/override the global variables/functions created in the webpage's JS world by its own scripts. There are ways to avoid that. At least you can limit the potential damage by packaging all your utilities into a single class/object. I believe there is a zero-pollution solution (nice rhyme!), too.

Reply:
@veloman-yunkan So the scripts are added to the UserWorld (see `getScript()` in kprofile.cpp ) which doesn't conflict with any of the webpage JS scripts.